app/validators/uniqueness.js

```diff
@@ -0,0 +1,57 @@
+import Ember from "ember";
+import Base from 'ember-validations/validators/base';
+
+export default Base.extend({
+  setupOptions: function() {
+    if(this.options === true) {
+      this.options = {};
+    }
+
+    let options = Ember.$.extend({
+      message: 'is taken.',
+      debounce: 150,
+      type: 'post',
+      data: {},
+    }, this.options);
+
+    this.set('options', options);
+  }.on('init'),
+
+  getDataPropertyName() {
+    return this.property.replace('model.', '');
+  },
+
+  call() {
+    if(this.model.get(this.property)) {
+      Ember.run.debounce(this, this.fetch, this.get('options.debounce'));
+    }
+  },
+
+  fetch() {
+    let options = this.get('options');
+    let errors = this.errors;
+    let message = this.options.message;
+    options.data[this.getDataPropertyName()] = this.model.get(this.property);
+
+    // Remove global error handler so it won't catch validation errors
+    let _onerror = Ember.onerror;
+    Ember.onerror = Ember.$.noop;
+
+    return new Ember.RSVP.Promise(function(resolve, reject) {
+      errors.clear();
+
+      return Ember.$.ajax(options)
```

Comment:
`Ember.$.ajax` is simply jquery's ajax (with its normal broken promises and lack of runloop). Please use https://github.com/aptible/diesel.aptible.com/blob/master/app/utils/ajax.js#L4